lib/kafka-consumer.js

```diff
@@ -52,26 +95,9 @@ function KafkaConsumer(conf, topicConf) {
 
   // If rebalance is undefined we don't want any part of this
   if (onRebalance && typeof onRebalance === 'boolean') {
-    conf.rebalance_cb = function(err, assignment) {
-      // Create the librdkafka error
-      err = LibrdKafkaError.create(err);
-      // Emit the event
-      self.emit('rebalance', err, assignment);
-
-      // That's it
-      try {
-        if (err.code === -175 /*ERR__ASSIGN_PARTITIONS*/) {
-          self.assign(assignment);
-        } else if (err.code === -174 /*ERR__REVOKE_PARTITIONS*/) {
-          self.unassign();
-        }
-      } catch (e) {
-        // Ignore exceptions if we are not connected
-        if (self.isConnected()) {
-          self.emit('rebalance.error', e);
-        }
-      }
-    };
+      conf.rebalance_cb = conf['partition.assignment.strategy'] === 'cooperative-sticky'
```

Comment:
Implementing and using [`rebalance_protocol` function](https://github.com/edenhill/librdkafka/blob/master/src-cpp/rdkafkacpp.h#L2895) to test for cooperative strategy would be better.

Reply:
It's seems to be impossible to do so as the return value is "NONE" until successfully joining